projects/packages/identity-crisis/src/class-rest-endpoints.php

```diff
@@ -74,6 +74,24 @@ public static function initialize_rest_api() {
 				'permission_callback' => array( static::class, 'url_secret_permission_check' ),
 			)
 		);
+
+		// Fetch URL verification secret.
+		register_rest_route(
+			'jetpack/v4',
+			'/identity-crisis/compare-url-secret',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
```

Comment:
Would it maybe make more sense for this to be a `GET` request? We are not modifying anything, right? Just comparing the secrets.

Reply:
Hi @fgiannar, thanks for taking a look!
The idea here is to keep the secret from showing up in the access log, which will happen if we use GET.
So it may be not be quite REST-ful, but safer that way 🙂